telegram/ext/dispatcher.py

```diff
@@ -183,11 +183,15 @@ def run_async(self, func, *args, **kwargs):
         self.__async_queue.put(promise)
         return promise
 
-    def start(self):
+    def start(self, ready=None):
         """Thread target of thread 'dispatcher'.
 
         Runs in background and processes the update queue.
 
+        Args:
+            ready (:obj:`threading.Event`, optional): If specified, the event will be set once the
+                dispatcher is ready.
+
         """
         if self.running:
```

Comment:
Shouldn't we do `ready.set()` here as well?

Reply:
that makes sense. i'll fix.